src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -294,16 +294,25 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
      *
      * @throws NotImplementedException
```

Comment:
The method still throws a NotImplementedException when the right side of the join is not an instance of QOM\SelectorInterface. Will add an explanation for the $root parameter!